packages/hyperion-autologging/src/ALSurfaceMutationPublisher.ts

```diff
@@ -56,33 +56,34 @@ export type InitOptions = Types.Options<{
   channel: ALSurfaceMutationChannel;
   flowletManager: ALFlowletManager;
   cacheElementReactInfo: boolean;
-  domSurfaceAttributeName?: string;
+  domSurfaceAttributeName: string;
   componentNameValidator?: ComponentNameValidator;
 }>;
 
+function validElement(node: Node | null): HTMLElement | null {
+  if (node != null && (node instanceof HTMLElement) && !/LINK|SCRIPT/.test(node.nodeName)) {
+    return node;
+  }
+  return null;
+}
+
 export function publish(options: InitOptions): void {
   const { domSurfaceAttributeName = AUTO_LOGGING_SURFACE, channel, flowletManager, cacheElementReactInfo, componentNameValidator = defaultComponentNameValidator } = options;
 
   setComponentNameValidator(componentNameValidator);
 
-  function processNode(node: Node, action: 'added' | 'removed') {
+  function processNode(node: HTMLElement | null, surface: string, action: 'added' | 'removed') {
     const timestamp = performanceAbsoluteNow();
 
     const flowlet = flowletManager.top();
-    if (!(node instanceof HTMLElement) || /LINK|SCRIPT/.test(node.nodeName)) {
-      return;
-    }
-    const surface = node.getAttribute(domSurfaceAttributeName);
```

Comment:
Generally this is a good change, however, I'm worried that if we allow this kind of breakage to happen, then the rest of the application may break. For example, when we have UI event, we will look for enclosing surface for that target element and we would need to see that surface. If we don't catch the problem here, we might have a harder time to see what was problem somewhere else.

Reply:
True,  also I just remembered,  if you just want simple surface mount/unmount the `al_surface_mount` and `al_surface_unmount` events are also available which this change basically re-implements/supports (same events just lacking the element info).  Will close this.